core/flamingock-test-support/src/main/java/io/flamingock/support/validation/impl/AuditSequenceStrictValidator.java

```diff
@@ -15,33 +15,99 @@
  */
 package io.flamingock.support.validation.impl;
 
+import io.flamingock.internal.common.core.audit.AuditEntry;
 import io.flamingock.internal.core.store.AuditStore;
 import io.flamingock.support.domain.AuditEntryDefinition;
 import io.flamingock.support.validation.SimpleValidator;
+import io.flamingock.support.validation.error.CountMismatchError;
+import io.flamingock.support.validation.error.ValidationError;
 import io.flamingock.support.validation.error.ValidationResult;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
+/**
+ * Validator that performs strict sequence validation of audit entries.
+ *
+ * <p>This validator verifies that the actual audit entries match the expected
+ * sequence exactly, both in count and in field values. Checking:</p>
+ * <ul>
+ *   <li>Exact count match between expected and actual entries</li>
+ *   <li>Strict field-by-field validation for each entry at each index</li>
+ *   <li>Order preservation (expected[0] must match actual[0], etc.)</li>
+ * </ul>
+ */
 public class AuditSequenceStrictValidator implements SimpleValidator {
 
     private static final String VALIDATOR_NAME = "Audit Sequence (Strict)";
 
-    private final AuditStore<?> auditStore;
-    private final List<AuditEntryExpectation> expectations;
-
+    private final List<AuditEntryExpectation> expectedExpectations;
+    private final List<AuditEntry> actualEntries;
 
+    /**
+     * Creates a strict sequence validator from an AuditStore and expected definitions.
+     *
+     * @param auditStore  the audit store to read actual entries from
+     * @param definitions the expected audit entry definitions
+     */
     public AuditSequenceStrictValidator(AuditStore<?> auditStore, AuditEntryDefinition... definitions) {
-        this.auditStore = auditStore;
-        this.expectations = Arrays.stream(definitions)
+        this(Arrays.asList(definitions), auditStore.getPersistence().getAuditHistory());
+    }
+
+    /**
+     * Internal constructor for direct list initialization (used by tests).
+     */
+    AuditSequenceStrictValidator(List<AuditEntryDefinition> expectedDefinitions, List<AuditEntry> actualEntries) {
+        this.expectedExpectations = expectedDefinitions.stream()
                 .map(AuditEntryExpectation::new)
                 .collect(Collectors.toList());
+        this.actualEntries = actualEntries != null ? actualEntries : new ArrayList<>();
     }
 
     @Override
     public ValidationResult validate() {
-        // TODO: Implement actual validation logic
-        return ValidationResult.success(VALIDATOR_NAME);
+        // Check count
+        if (expectedExpectations.size() != actualEntries.size()) {
+            return ValidationResult.failure(VALIDATOR_NAME,
```

Comment:
We should fail fast. This should be added to the list of errors and continue checking other errors.

The following logic should be placed in the loop, but this is the summary:
- If the expectedEntries.size() > actualEntries.size() -> for each entry it should add an MissingEntryError with the details of the missing entry
- If the expectedEntries.size() < actualEntries.size() -> for each entry it should add an UnexpectedEntryError with the details of the missing entry